src/hyperloglog/mod.rs

```diff
@@ -114,14 +120,27 @@ where
 {
     /// Same as `new` but with a specific `BuildHasher`.
     pub fn with_hash(b: usize, buildhasher: B) -> Self {
+        let m = 1_usize << b;
+        let registers = vec![0; m];
+        Self::with_registers_and_hash(b, registers, buildhasher)
+    }
+
+    /// Same as `new` but with pre-initialized registers and a specific `BuildHasher`.
+    pub fn with_registers_and_hash(b: usize, registers: Vec<u8>, buildhasher: B) -> Self {
```

Comment:
Why is this required? As far as I can see, the `serde` impl doesn't need that. Also there's no way to actually read the registers, so this seems rather useless.

Reply:
I've added the corresponding `registers()` [here](https://github.com/crepererum/pdatastructs.rs/pull/129/files#diff-37372a8fbd5759f7e6fef0e85f6ffa4783c80319a4a6a3806009270dbfa63578R165-R167). 

As I mentioned in the other replies to your comments, what I actually need to do is to serialize this structure using Protobuf. The initial design was to introduce an auxiliary structure that would decouple the actual HLL parameters (`b` and `registers`) from the `PhantomData` and the `T` as those are only needed for hashing and not actually relevant to the HLL (if we view the HLL as if it was working with already hashed values).

However the requirement to implement `serde` for the `HyperLogLog` structure directly leaves me without the ability to do the protobuf serialization. So I introduced a method to both retrieve the `registers` and `b`, as well as a new constructor that takes these 2 params along with a hasher.

Reply:
Alright, sounds good to me then.